src/control/Control.cpp

```diff
@@ -370,7 +370,7 @@ void Control::updatePageNumbers(size_t page, size_t pdfPage) {
     this->win->updatePageNumbers(page, this->doc->getPageCount(), pdfPage);
     this->sidebar->selectPageNr(page, pdfPage);
 
-    this->metadata->storeMetadata(this->doc->getEvMetadataFilename().str(), page, getZoomControl()->getZoomReal());
+    this->metadata->storeMetadata(this->doc->getEvMetadataFilename().string(), page, getZoomControl()->getZoomReal());
```

Comment:
Should fs::path then be used everywhere in the MetadataManager?

Reply:
Right, just looked into it, and it uses the string as input for the glib file api. So either it must be utf8 (which is really unclear and intransperant until c++20 (they introduced a u8char and a u8string **finally**)) or we bypass it with fs::path. Which is even better, since the path object is modified in the MetaDataManager

Reply:
I've also noticed that fs::path doesn't cast implicitly to string on Windows. So either I reintroduce `file.string()` in many function calls or I make sure those functions take a proper fs::path since a file path is what they want. The latter is obviously a much better approach, so I'll be looking into that.